### PR TITLE
Add GitHub Actions CI workflow

### PR DESCRIPTION
**Pull request by RohanMenon**
Travis CI to GHA migration. Issue raised here: https://github.com/prestosql/presto/issues/5256

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: ['1.8', '11', '15']
```
Comment (Praveen2112) [inline .github/workflows/main.yml]:
```suggestion
        java: ['8', '11', '15']
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        java-version: ${{ matrix.java }}
+
+    - name: Maven Install Dependencies
+      run: mvn install -B -V -DskipTests -Dair.check.skip-all
+    - name: Maven Offline
+      run: mvn dependency:go-offline -B -Dsilent
+    - name: Run On Docker
+      run: mvn install -B
+            - ./tempto-examples/bin/run_on_docker.sh --thread-count 2 -x failing -e io.prestosql.tempto.examples.ExclusionTest.failingTest,io.prestosql.tempto.examples.AnotherExclusionTest
+            - CONFIG_FILE=tempto-configuration-read-only.yaml ./tempto-examples/bin/run_on_docker.sh --thread-count 2 -g in_memory
```
Comment (Praveen2112) [inline .github/workflows/main.yml]: @findepi Is it okay if we could capture these also as `matrix` like for `CONFIG_FILE` and `g` as (in_memory and ssh)

**Review by Praveen2112 — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: .github/workflows/main.yml (hunk 2 above)
Comment (findepi) [reply .github/workflows/main.yml]: if they are short, it's OK to run them sequentially

**Comment by findepi** [comment]
@RohanMenon thanks for your PR!

small note: in Presto we settled on the file name `.github/workflows/ci.yml`. Let's use the same here

**Comment by Praveen2112** [comment]
Commit message can be like Add GitHub Workflow

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: CI
```
Comment (Praveen2112) [inline .github/workflows/ci.yml:1]:
```suggestion
name: ci
```

**Review by Praveen2112 — APPROVED** [review]
Can you squash the commits.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      with:
+        java-version: ${{ matrix.java }}
+
+    - name: Maven Install Dependencies
+      run: mvn install -B -V -DskipTests -Dair.check.skip-all
+    - name: Maven Offline
+      run: mvn dependency:go-offline -B -Dsilent
+    - name: Run On Docker
+      run: mvn install -B
+            - ./tempto-examples/bin/run_on_docker.sh --thread-count 2 -x failing -e io.prestosql.tempto.examples.ExclusionTest.failingTest,io.prestosql.tempto.examples.AnotherExclusionTest
```
Comment (Praveen2112) [inline .github/workflows/ci.yml:25]: We need to add `run` for each of these lines

**Comment by Praveen2112** [comment]
@RohanMenon Gentle ping

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/checkout@v2
+    - name: ${{ matrix.java }}
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ matrix.java }}
+
+    - name: Maven Install Dependencies
+      run: mvn install -B -V -DskipTests -Dair.check.skip-all
+    - name: Maven Offline
+      run: mvn dependency:go-offline -B -Dsilent
```
Comment (findepi) [inline .github/workflows/ci.yml:22]: Why this?

cc @wendigo

Comment (findepi) [reply .github/workflows/ci.yml:22]: Would that solve https://github.com/prestosql/presto/issues/5816 ?

Comment (wendigo) [reply .github/workflows/ci.yml:22]: We can try that @findepi

**Comment by Praveen2112** [comment]
@RohanMenon Any updates on this ?

**Comment by kokosing** [comment]
Can you please rebase?

**Comment by Praveen2112** [comment]
This is implemented by this PR - https://github.com/trinodb/tempto/pull/83

**Pull request closed without merging**